### PR TITLE
adds boilerplate cli for chats backup

### PR DESCRIPTION
**Pull request by ryanfkeepers**
All code is copied and amended from existing cli boilerplate.
CLI commands are marked PreRelease at this time.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #5062

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by HiteshRepo — APPROVED** [review]

File: src/cli/backup/teamschats.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	"github.com/alcionai/corso/src/pkg/services/m365"
+)
+
+// ------------------------------------------------------------------------------------------------
+// setup and globals
+// ------------------------------------------------------------------------------------------------
+
+const (
+	teamschatsServiceCommand                 = "chats"
+	teamschatsServiceCommandCreateUseSuffix  = "--user <userEmail> | '" + flags.Wildcard + "'"
```
Comment (pandeyabs) [inline src/cli/backup/teamschats.go:28]: What's the `|` for?

File: src/cli/backup/teamschats.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+// ------------------------------------------------------------------------------------------------
+// backup create
+// ------------------------------------------------------------------------------------------------
+
+// `corso backup create chats [<flag>...]`
+func teamschatsCreateCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:     teamschatsServiceCommand,
+		Aliases: []string{teamsServiceCommand},
+		Short:   "Backup M365 Chats service data",
```
Comment (pandeyabs) [inline src/cli/backup/teamschats.go:105]:
```suggestion
		Short:   "Backup M365 Teams Chats service data",
```

File: src/cli/backup/teamschats.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+// ------------------------------------------------------------------------------------------------
+// backup list
+// ------------------------------------------------------------------------------------------------
+
+// `corso backup list teamschats [<flag>...]`
+func teamschatsListCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   teamschatsServiceCommand,
+		Short: "List the history of M365 TeamsChats service backups",
```
Comment (pandeyabs) [inline src/cli/backup/teamschats.go:173]: nit: We are using `TeamsChats` for CLI command descriptions. Should we change these to `Teams Chats`, to be consistent with MSFT terminology?

```suggestion
		Short: "List the history of M365 Teams Chats service backups",
```

**Review by pandeyabs — APPROVED** [review]
🚀

File: src/cli/backup/teamschats.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/cli/backup/teamschats.go:28]: to indicate that --user takes either value.

File: src/cli/backup/teamschats.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/cli/backup/teamschats.go:105]: `teamschats` is sort of an internal moniker.  On the end UX we're going to just call it `chats`.

File: src/cli/backup/teamschats.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/cli/backup/teamschats.go:173]: hah, that was a find-replace mistake, and should only say `Chats`

**Pull request merged**